src/components/EditableCard.tsx

```diff
@@ -20,7 +21,9 @@ import FormModal from './FormModal';
 
 export interface IEditableCardProps extends ICardProps, ISharedFormProps {
   ModalComponent?: new (props: ISharedFormModalProps) => FormModal | FormDrawer;
+  deleteDisabledTooltip?: string;
   onDelete?: (model: unknown) => Promise<any>;
+  protectedField?: string | null;
```

Comment:
I don't think that's possible because the frontend just passes in the `model` to `fields-ant`, then `fields-ant` creates an `EditableCard` of each instance of the model in the `EditableArrayCard` component. Each instance could have or not have payments.

Reply:
Hmm I'm not quite sure I'm understanding. Each investment could have or not have `payments`, so I have to pass the field along with the `model` to `EditableArrayCard` so when it renders each `EditableCard` it can check whether that instance has payments or not. From inside `coop-client` I can't pass a boolean for every investment since the frontend only passes an array. Let me know if I'm understanding you correctly.

Reply:
In the future we could do a `disableDelete` for permissions since we could just disable the delete button for the entire `EditableArrayCard`

Reply:
this prop feels way too specific for the component i think. Do you think this is possible?
```
<EditableArrayCard
...
renderDelete={(investment) => investment.payments && item.payments.length > 0}
...
/>
```

```
const enableDelete = renderDelete ? renderDelete(model) : false;
```

something like that... then we can just pass `(investment) => false` to the prop?

Reply:
also didnt realize that this was a array card! but i think that might work?!

Reply:
yeah agreed! I think passing the function is much better. updated!